Author: Jamie Strandboge <jamie.strandboge@ubuntu.com>
Description: Adjust build system to build what we want
 The upstream build system doesn't have particularl good support for
 configuring our target via commandline options, so we instead patch the
 generated files.
Index: edk2-0~20121205.edae8d2d/Conf/target.txt
===================================================================
--- edk2-0~20121205.edae8d2d.orig/Conf/target.txt
+++ edk2-0~20121205.edae8d2d/Conf/target.txt
@@ -23,7 +23,9 @@
 #                                               build. This line is required if and only if the current
 #                                               working directory does not contain one or more description
 #                                               files.
-ACTIVE_PLATFORM       = Nt32Pkg/Nt32Pkg.dsc
+#ACTIVE_PLATFORM       = MdeModulePkg/MdeModulePkg.dsc
+ACTIVE_PLATFORM       = OvmfPkg/OvmfPkgX64.dsc
+ 
 
 #  TARGET                List       Optional    Zero or more of the following: DEBUG, RELEASE, NOOPT
 #                                               UserDefined; separated by a space character.
@@ -46,7 +48,7 @@
 #                                               If not specified, then all valid architectures specified
 #                                               in the platform file, for which tools are available, will be
 #                                               built.
-TARGET_ARCH           = IA32
+TARGET_ARCH           = X64
 
 #  TOOL_DEFINITION_FILE  Filename  Optional   Specify the name of the filename to use for specifying
 #                                             the tools to use for the build.  If not specified,
@@ -56,12 +58,12 @@
 #  TAGNAME               List      Optional   Specify the name(s) of the tools_def.txt TagName to use.
 #                                             If not specified, all applicable TagName tools will be
 #                                             used for the build.  The list uses space character separation.
-TOOL_CHAIN_TAG        = MYTOOLS
+TOOL_CHAIN_TAG        = GCC47
 
 # MAX_CONCURRENT_THREAD_NUMBER  NUMBER  Optional  The number of concurrent threads. Recommend to set this
 #                                                 value to one more than the number of your compurter
 #                                                 cores or CPUs. Less than 2 means disable multithread build.
-MAX_CONCURRENT_THREAD_NUMBER = 1
+MAX_CONCURRENT_THREAD_NUMBER = 5
 
 
 # BUILD_RULE_CONF  Filename Optional  Specify the file name to use for the build rules that are followed
Index: edk2-0~20121205.edae8d2d/Conf/tools_def.txt
===================================================================
--- edk2-0~20121205.edae8d2d.orig/Conf/tools_def.txt
+++ edk2-0~20121205.edae8d2d/Conf/tools_def.txt
@@ -147,6 +147,9 @@
 DEFINE GCC46_IA32_PREFIX       = /usr/bin/
 DEFINE GCC46_X64_PREFIX        = /usr/bin/
 
+DEFINE GCC47_IA32_PREFIX       = /usr/bin/
+DEFINE GCC47_X64_PREFIX        = /usr/bin/
+
 DEFINE UNIX_IASL_BIN           = /usr/bin/iasl
 #DEFINE UNIX_IASL_BIN           = $(HOME)/programs/iasl
 DEFINE WIN_ASL_BIN_DIR         = C:\ASL
@@ -277,6 +280,12 @@
 #                             Required to build platforms or ACPI tables:
 #                               Intel(r) ACPI Compiler v20101013 from
 #                               http://www.acpica.org/downloads/previous_releases.php
+#   GCC47       -Linux-  Requires:
+#                             GCC 4.7 (Native)
+#                        Optional:
+#                             Required to build platforms or ACPI tables:
+#                               Intel(r) ACPI Compiler v20101013 from
+#                               http://www.acpica.org/downloads/previous_releases.php
 #   ELFGCC      -Linux-  Requires:
 #                             GCC(this tool chain uses whatever version of gcc and binutils that is installed in /usr/bin)
 #                        Optional:
@@ -2584,6 +2593,14 @@
 DEFINE GCC46_X64_DLINK_FLAGS         = DEF(GCC45_X64_DLINK_FLAGS)
 DEFINE GCC46_ASM_FLAGS               = DEF(GCC45_ASM_FLAGS)
 
+DEFINE GCC47_IA32_CC_FLAGS           = DEF(GCC45_IA32_CC_FLAGS) -Wno-address -Wno-unused-but-set-variable
+DEFINE GCC47_X64_CC_FLAGS            = DEF(GCC45_X64_CC_FLAGS) -Wno-address -Wno-unused-but-set-variable
+DEFINE GCC47_IA32_X64_DLINK_COMMON   = DEF(GCC45_IA32_X64_DLINK_COMMON)
+DEFINE GCC47_IA32_X64_ASLDLINK_FLAGS = DEF(GCC45_IA32_X64_ASLDLINK_FLAGS)
+DEFINE GCC47_IA32_X64_DLINK_FLAGS    = DEF(GCC45_IA32_X64_DLINK_FLAGS)
+DEFINE GCC47_X64_DLINK_FLAGS         = DEF(GCC45_X64_DLINK_FLAGS)
+DEFINE GCC47_ASM_FLAGS               = DEF(GCC45_ASM_FLAGS)
+
 ####################################################################################
 #
 # Unix GCC And Intel Linux ACPI Compiler
@@ -2867,6 +2884,71 @@
 
 ####################################################################################
 #
+# GCC 4.7 - This configuration is used to compile under Linux to produce
+#           PE/COFF binaries using GCC 4.7.
+#
+####################################################################################
+*_GCC47_*_*_FAMILY               = GCC
+
+*_GCC47_*_MAKE_PATH                    = make
+*_GCC47_*_ASL_PATH                     = DEF(UNIX_IASL_BIN)
+
+*_GCC47_*_PP_FLAGS                     = DEF(GCC_PP_FLAGS)
+*_GCC47_*_ASLPP_FLAGS                  = DEF(GCC_ASLPP_FLAGS)
+*_GCC47_*_ASLCC_FLAGS                  = DEF(GCC_ASLCC_FLAGS)
+*_GCC47_*_VFRPP_FLAGS                  = DEF(GCC_VFRPP_FLAGS)
+*_GCC47_*_APP_FLAGS                    =
+*_GCC47_*_ASL_FLAGS                    = DEF(IASL_FLAGS)
+*_GCC47_*_ASL_OUTFLAGS                 = DEF(IASL_OUTFLAGS)
+
+##################
+# GCC47 IA32 definitions
+##################
+*_GCC47_IA32_OBJCOPY_PATH         = DEF(GCC47_IA32_PREFIX)objcopy
+*_GCC47_IA32_CC_PATH              = DEF(GCC47_IA32_PREFIX)gcc
+*_GCC47_IA32_SLINK_PATH           = DEF(GCC47_IA32_PREFIX)ar
+*_GCC47_IA32_DLINK_PATH           = DEF(GCC47_IA32_PREFIX)ld
+*_GCC47_IA32_ASLDLINK_PATH        = DEF(GCC47_IA32_PREFIX)ld
+*_GCC47_IA32_ASM_PATH             = DEF(GCC47_IA32_PREFIX)gcc
+*_GCC47_IA32_PP_PATH              = DEF(GCC47_IA32_PREFIX)gcc
+*_GCC47_IA32_VFRPP_PATH           = DEF(GCC47_IA32_PREFIX)gcc
+*_GCC47_IA32_ASLCC_PATH           = DEF(GCC47_IA32_PREFIX)gcc
+*_GCC47_IA32_ASLPP_PATH           = DEF(GCC47_IA32_PREFIX)gcc
+*_GCC47_IA32_RC_PATH              = DEF(GCC47_IA32_PREFIX)objcopy
+
+*_GCC47_IA32_ASLCC_FLAGS          = DEF(GCC_ASLCC_FLAGS) -m32
+*_GCC47_IA32_ASLDLINK_FLAGS       = DEF(GCC47_IA32_X64_ASLDLINK_FLAGS) -m elf_i386
+*_GCC47_IA32_ASM_FLAGS            = DEF(GCC47_ASM_FLAGS) -m32 -march=i386
+*_GCC47_IA32_CC_FLAGS             = DEF(GCC47_IA32_CC_FLAGS) -Os
+*_GCC47_IA32_DLINK_FLAGS          = DEF(GCC47_IA32_X64_DLINK_FLAGS) -m elf_i386 --oformat=elf32-i386
+*_GCC47_IA32_RC_FLAGS             = DEF(GCC_IA32_RC_FLAGS)
+*_GCC47_IA32_OBJCOPY_FLAGS        = 
+
+##################
+# GCC47 X64 definitions
+##################
+*_GCC47_X64_OBJCOPY_PATH         = DEF(GCC47_X64_PREFIX)objcopy
+*_GCC47_X64_CC_PATH              = DEF(GCC47_X64_PREFIX)gcc
+*_GCC47_X64_SLINK_PATH           = DEF(GCC47_X64_PREFIX)ar
+*_GCC47_X64_DLINK_PATH           = DEF(GCC47_X64_PREFIX)ld
+*_GCC47_X64_ASLDLINK_PATH        = DEF(GCC47_X64_PREFIX)ld
+*_GCC47_X64_ASM_PATH             = DEF(GCC47_X64_PREFIX)gcc
+*_GCC47_X64_PP_PATH              = DEF(GCC47_X64_PREFIX)gcc
+*_GCC47_X64_VFRPP_PATH           = DEF(GCC47_X64_PREFIX)gcc
+*_GCC47_X64_ASLCC_PATH           = DEF(GCC47_X64_PREFIX)gcc
+*_GCC47_X64_ASLPP_PATH           = DEF(GCC47_X64_PREFIX)gcc
+*_GCC47_X64_RC_PATH              = DEF(GCC47_X64_PREFIX)objcopy
+
+*_GCC47_X64_ASLCC_FLAGS          = DEF(GCC_ASLCC_FLAGS) -m64
+*_GCC47_X64_ASLDLINK_FLAGS       = DEF(GCC47_IA32_X64_ASLDLINK_FLAGS) -melf_x86_64
+*_GCC47_X64_ASM_FLAGS            = DEF(GCC47_ASM_FLAGS) -m64 -Wl,-melf_x86_64
+*_GCC47_X64_CC_FLAGS             = DEF(GCC47_X64_CC_FLAGS)
+*_GCC47_X64_DLINK_FLAGS          = DEF(GCC47_X64_DLINK_FLAGS)
+*_GCC47_X64_RC_FLAGS             = DEF(GCC_X64_RC_FLAGS)
+*_GCC47_X64_OBJCOPY_FLAGS        = 
+
+####################################################################################
+#
 # Cygwin GCC And Intel ACPI Compiler
 #
 ####################################################################################
